src/main/java/org/folio/oaipmh/service/impl/SetServiceImpl.java

```diff
@@ -42,4 +75,76 @@ public SetServiceImpl(final SetDao setDao) {
     return setDao.deleteSetById(id, tenantId);
   }
 
+  @Override
+  public Future<FilteringConditionValueCollection> getFilteringConditions(Map<String, String> okapiHeaders) {
+    Promise<FilteringConditionValueCollection> promise = Promise.promise();
+    HttpClient vertxHttpClient = Vertx.vertx()
+      .createHttpClient();
+    Future<JsonObject> locationFuture = getFilteringConditionValues(LOCATION_URI, vertxHttpClient, okapiHeaders);
+    Future<JsonObject> illPoliciesFuture = getFilteringConditionValues(ILL_POLICIES_URI, vertxHttpClient, okapiHeaders);
+    Future<JsonObject> materialTypesFuture = getFilteringConditionValues(MATERIAL_TYPES_URI, vertxHttpClient, okapiHeaders);
+    Future<JsonObject> resourceTypeFuture = getFilteringConditionValues(RESOURCE_TYPES_URI, vertxHttpClient, okapiHeaders);
+    Future<JsonObject> instanceFormatsFuture = getFilteringConditionValues(INSTANCE_FORMATS_URI, vertxHttpClient, okapiHeaders);
+
+    CompositeFuture.all(locationFuture, illPoliciesFuture, materialTypesFuture, resourceTypeFuture, instanceFormatsFuture)
+      .onComplete(result -> {
+        if (result.failed()) {
+          promise.fail(result.cause());
+        } else {
+          List<SetsFilteringCondition> values = new ArrayList<>();
+          values.add(jsonObjectToSetsFilteringCondition(locationFuture.result(), LOCATION_JSON_FIELD_PATH, LOCATION));
+          values.add(jsonObjectToSetsFilteringCondition(illPoliciesFuture.result(), ILL_POLICIES_JSON_FIELD_PATH, ILL_POLICIES));
+          values
+            .add(jsonObjectToSetsFilteringCondition(materialTypesFuture.result(), MATERIAL_TYPES_JSON_FIELD_PATH, MATERIAL_TYPES));
+          values
+            .add(jsonObjectToSetsFilteringCondition(resourceTypeFuture.result(), INSTANCE_TYPES_JSON_FIELD_PATH, INSTANCE_TYPES));
+          values.add(jsonObjectToSetsFilteringCondition(instanceFormatsFuture.result(), INSTANCE_FORMATS_JSON_FIELD_PATH,
+              INSTANCE_FORMATS));
+
+          FilteringConditionValueCollection filteringConditionValueCollection = new FilteringConditionValueCollection()
+            .withSetsFilteringConditions(values);
+          promise.complete(filteringConditionValueCollection);
+        }
+      });
+    return promise.future();
+  }
+
+  private Future<JsonObject> getFilteringConditionValues(String requestUri, HttpClient httpClient,
+      Map<String, String> okapiHeaders) {
+    Promise<JsonObject> promise = Promise.promise();
+
+    String okapiUrl = okapiHeaders.get(OKAPI_URL);
+    String tenant = okapiHeaders.get(OKAPI_TENANT);
+    String token = okapiHeaders.get(OKAPI_TOKEN);
+
+    HttpClientRequest httpClientRequest = httpClient.getAbs(okapiUrl.concat(requestUri))
+      .putHeader(OKAPI_TOKEN, token)
+      .putHeader(OKAPI_TENANT, tenant)
+      .putHeader(ACCEPT, APPLICATION_JSON);
+
+    httpClientRequest.handler(response -> {
+      if (response.statusCode() == 200)
```

Comment:
This is personal preference - I'm not sure if there's a project-wide standard, but I prefer to always use brackets for conditionals, even in cases like this.

Reply:
Thank you for advice, i've added them here.